src/routes/safe/components/Transactions/TxList/BatchExecute.tsx

```diff
@@ -0,0 +1,240 @@
+      dispatch,
+      account,
+      safeAddress,
+    })
+
+    toggleModal()
+  }
+
+  return (
+    <>
+      <StyledButton
+        color="primary"
+        variant="contained"
+        onClick={handleOpenModal}
+        disabled={!isBatchable || hasPendingTx}
+        onMouseEnter={handleOnMouseEnter}
+        onMouseLeave={handleOnMouseLeave}
+      >
+        Execute Batch {isBatchable ? `(${batchableTransactions.length})` : ''}
+      </StyledButton>
+      <Modal description="Execute Batch" handleClose={toggleModal} open={isModalOpen} title="Execute Batch">
+        <ModalHeader onClose={toggleModal} title="Batch-Execute transactions" />
+        <Hairline />
+        <ModalContent>
+          <Row margin="md">
+            <Paragraph noMargin>
+              This transaction batches a total of {batchableTransactions.length} transactions from your queue into a
+              single Ethereum transaction. Please check every included transaction carefully, especially if you have
+              rejection transactions, and make sure you want to execute all of them. Included transactions are
+              highlighted in green when you hover over the execute button.
+            </Paragraph>
+          </Row>
+          <Row margin="xs">
+            <Paragraph color="disabled" noMargin size="lg">
+              {getInteractionTitle('0')}
+            </Paragraph>
+          </Row>
+          <Row margin="md">
+            <PrefixedEthHashInfo
+              hash={multiSendContractAddress}
+              showAvatar
+              textSize="lg"
+              showCopyBtn
+              explorerUrl={getExplorerInfo(multiSendContractAddress)}
+            />
+          </Row>
+          <Row margin="md">
+            {batchableTransactions.map((transaction) => {
+              if (!transaction.txDetails?.txData) return null
+
+              const tx = {
+                to: getTxRecipient(transaction.txInfo, safeAddress),
+                value: getTxValue(transaction.txInfo, transaction.txDetails),
+                data: transaction.txDetails?.txData.hexData || EMPTY_DATA,
+              }
+              const decodedDataParams: DecodedDataParameterValue = {
+                operation: 0,
+                to: tx.to,
+                value: tx.value,
+                data: transaction.txDetails?.txData.hexData || EMPTY_DATA,
+                dataDecoded: null,
+              }
+
+              const decodedData = transaction.txDetails?.txData.dataDecoded || decodedDataParams
+
+              return <DecodeTxs txs={[tx]} decodedData={decodedData as DecodedTxDetailType} key={transaction.id} />
+            })}
+          </Row>
+          <Paragraph size="md" align="center" color="disabled" noMargin>
+            This feature is still in experimental mode. Be aware, that if one of the included transactions fails, all
+            included transactions will fail. This will result in the loss of the allocated transaction fees.
```

Comment:
If you definitely lose your fees, you need update the settings text to use this sentence instead.

Reply:
The Settings text also says "...loss of the allocated transaction fees."

Reply:
"risking the loss" is not definitive. I would use this sentence there.

Reply:
Copied the text from the Modal to the Settings page.